pkg/api/customization/catalog/store.go

```diff
@@ -0,0 +1,29 @@
+package catalog
+
+import (
+	"fmt"
+
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+)
+
+type Store struct {
+	types.Store
+}
+
+func Wrap(store types.Store) types.Store {
+	return &Store{
+		store,
+	}
+}
+
+func (s *Store) Delete(apiContext *types.APIContext, schema *types.Schema, id string) (map[string]interface{}, error) {
+	catalog, err := s.ByID(apiContext, schema, id)
+	if err != nil {
+		return nil, err
+	}
+	if catalog["url"] == systemLibraryURL && catalog["name"] == "system-library" {
+		return nil, httperror.NewAPIError(httperror.Conflict, fmt.Sprint("[catalog store] Unathorized to delete system-library catalog"))
```

Comment:
The error should be a `InvalidBodyContent`, `conflict` would most likely apply to a PUT request. The message itself should be updated to `not allowed to delete system-library`